Source/WebCore/rendering/RenderVideo.cpp

```diff
@@ -175,6 +175,11 @@ void RenderVideo::imageChanged(WrappedImagePtr newImage, const IntRect* rect)
     updateIntrinsicSize();
 }
 
+static bool areAspectRatiosEssentiallyEqual(const auto& size1, const auto& size2)
+{
+    return fabs(size1.aspectRatio() - size2.aspectRatio()) <= 0.01f;
```

Comment:
I know you explained in the commit message why this is reasonable for comparing aspect ratios, but would WTF::areEssentiallyEqual work here instead?

Reply:
Oh thanks I didn't know about `WTF::areEssentiallyEqual`. I could pass an epsilon to that as the default of `std::numeric_limits<T>::epsilon()` is too small.

The offset from UIKit and the video will be at most one point so I could pass `page().deviceScaleFactor()` and write the test like so:

```
return fabs(size1.width() - size2.width()) <= deviceScaleFactor &&
     fabs(size1.height() - size2.height()) <= deviceScaleFactor;
```
which is more precise.

Reply:
I updated the PR to use WTF::areEssentiallyEqual and compute the tolerance based on the deviceScaleFactor and the content size of the layer in fullscreen. This is a more accurate check.